pkg/core/statesync/neotest_test.go

```diff
@@ -6,6 +6,7 @@ import (
 
 	"github.com/nspcc-dev/neo-go/internal/basicchain"
 	"github.com/nspcc-dev/neo-go/pkg/config"
+	"github.com/nspcc-dev/neo-go/pkg/core"
```

Comment:
Create a test for `SetOnStageChanged`, some simple callback is enough, use a counter that will be incremented every time when state is changed, ensure that callback is called expected number of times.